src/compiler/program.ts

```diff
@@ -4039,7 +4040,7 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
 
                 const isFromNodeModulesSearch = resolution.isExternalLibraryImport;
                 const isJsFile = !resolutionExtensionIsTSOrJson(resolution.extension);
-                const isJsFileFromNodeModules = isFromNodeModulesSearch && isJsFile;
+                const isJsFileFromNodeModules = isFromNodeModulesSearch && isJsFile && (!resolution.originalPath || pathContainsNodeModules(resolution.resolvedFileName));

```

Comment:
Do we want to process this file as if it was not found in `node_modules` if yes i dont think it is enough. We increase `currentNodeModulesDepth` so we will add this file but may still elide its impoirts?

Reply:
I pushed some test changes that I think show it’s working.

Reply:
But with `allowJS`, `maxNodeModuleJsDepth` is 2 so you would need to adjust that in the test case right ?

Reply:
No, it only defaults to 2 with a jsconfig.json as opposed to a tsconfig.json.